### DISCUSSION TITLE
End of Line character management - .gitattributes ?

### DISCUSSION THREAD

**Discussion by ChrisNolan** [karrioapi/karrio#640] [Ideas]
I did my first modification to the source today - go me! 🎉 

One of the challenges I experienced setting up my development environment (following [this document](https://docs.karrio.io/product/resources/development/#developing-with-docker) ) was my local machine by default checked out the scripts with 'CRLF', but when I went to run those scripts inside of the docker container, it errored because of the "\r"s that were in there.

I was able to resolve this by `git config core.autocrlf input` and re-checking out the project.

How do people feel though about setting up a ".gitattributes" file that could 'force' these types of settings on the project as a whole?
This is 'new' to me, but when i was looking into how to resolve my issue it was suggested that using this file was a common practice?  Some [templates are here](https://github.com/gitattributes/gitattributes) if you'd like to review.

Here is a [blog post about .gitattributes](https://dev.to/deadlybyte/please-add-gitattributes-to-your-git-repository-1jld) for consideration as well.

**Comment by danh91** [comment]
I'm not too familiar with that but if it helps, I'm all for it.
